core/ssh_keys_test.go

```diff
@@ -0,0 +1,476 @@
+package core
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+	"testing"
+
+	"github.com/krystal/go-katapult"
+	"github.com/krystal/go-katapult/internal/test"
+	"github.com/krystal/go-katapult/internal/testclient"
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_AuthSSHKey_JSONMarshaling(t *testing.T) {
+	tests := []struct {
+		name string
+		obj  *AuthSSHKey
+	}{
+		{
+			name: "empty",
+			obj:  &AuthSSHKey{},
+		},
+		{
+			name: "full",
+			obj: &AuthSSHKey{
+				ID:          "a",
+				Name:        "b",
+				Fingerprint: "c",
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			testJSONMarshaling(t, tt.obj)
+		})
+	}
+}
+
+func Test_sshKeyResponseBody_JSONMarshaling(t *testing.T) {
+	tests := []struct {
+		name string
+		obj  *sshKeysResponseBody
+	}{
+		{
+			name: "empty",
+			obj:  &sshKeysResponseBody{},
+		},
+		{
+			name: "full",
+			obj: &sshKeysResponseBody{
+				Pagination: &katapult.Pagination{
+					CurrentPage: 33,
+				},
+				SSHKey:  &AuthSSHKey{ID: "1"},
+				SSHKeys: []*AuthSSHKey{{ID: "2"}},
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			testJSONMarshaling(t, tt.obj)
+		})
+	}
+}
+
+func Test_AuthSSHKeyProperties_JSONMarshaling(t *testing.T) {
+	tests := []struct {
+		name string
+		obj  *AuthSSHKey
+	}{
+		{
+			name: "empty",
+			obj:  &AuthSSHKey{},
```

Comment:
Apologies for not spotting this earlier, but this test should be testing the `AuthSSHKeyProperties` struct, not `AuthSSHKey` :)